src/main/java/step2/StringCalculatorVO.java

```diff
@@ -0,0 +1,66 @@
+package step2;
+
+import step2.enums.RegexPattern;
+
+import java.util.Arrays;
+
+import static step2.enums.RegexPattern.*;
+
+public class StringCalculatorVO {
+
+    private final Integer[] numberElements;
+    private final String[] operatorElements;
+
+    private final StringCalculatorVOValidator stringCalculatorVOValidator = new StringCalculatorVOValidator();
+
+    public StringCalculatorVO(String expression) {
+        String[] splitElements = splitStringByWhiteSpace(expression);
+        String[] operatorElements = filterOperatorStringFromStringArray(splitElements);
+        Integer[] numberElements = transformStringArrayToIntegerArray(filterNumberStringFromStringArray(splitElements));
+
+        stringCalculatorVOValidator.validateIsNotNullOrEmpty(expression)
+                                .validateExpressionStartOrEndNumber(splitElements)
+                                .validateOperatorPlacedBetweenNumbers(splitElements)
+                                .validateOperatorCount(operatorElements.length, numberElements.length);
+
+        this.numberElements = numberElements;
+        this.operatorElements = operatorElements;
+    }
+
+    protected StringCalculatorVO() {
+        this.numberElements = null;
+        this.operatorElements = null;
+    }
+
+    public Integer[] getNumberElements() {
+        return numberElements;
+    }
+
+    public String[] getOperatorElements() {
+        return operatorElements;
+    }
+
+    protected String[] splitStringByWhiteSpace(String expression) {
+        return WHITE_SPACE_REGEX.split(expression);
+    }
+
+    protected String[] filterNumberStringFromStringArray(String[] stringArray) {
```

Comment:
전체적으로 메소드 이름이 장황하다는 느낌이 듭니다 :)
인수와 반환타입으로 유추할 수 있는 부분은 생략하는게 어떨까요?
다른 메소드들도 마찬가지입니다!
```suggestion
    protected String[] filterNumbers(String[] stringArray) {
```